### PR TITLE
Add hashdeep 4.4 package

### PR DESCRIPTION
**Pull request by karantan**
A set of cross-platform tools to compute hashes.

See: http://md5deep.sourceforge.net/start-hashdeep.html

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
      Fails with
```
...
In file included from main.cpp:16:
In file included from ./main.h:23:
./xml.h:103:82: error: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wreserved-user-defined-literal]
    void xmlout(const std::string &tag,const int64_t value){ xmlprintf(tag,"","%"PRId64,value); }
```
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by karantan** [comment]
My first PR (for nix) so I might have missed something.

**Comment by domenkozar** [comment]
@GrahamcOfBorg build hashdeep

File: pkgs/tools/security/hashdeep/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "0m2b042ndikavmplv3qjdhfj44hl1h8car83c192xi9nv5ahi7mf";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  meta = with stdenv.lib; {
+    description = "A set of cross-platform tools to compute hashes.";
+    homepage = "https://github.com/jessek/hashdeep";
+    license = licenses.gpl2;
+    platforms = with platforms; linux ++ freebsd ++ openbsd ++ darwin;
```
Comment (domenkozar) [inline pkgs/tools/security/hashdeep/default.nix]:
```suggestion
    platforms = with platforms; linux ++ freebsd ++ openbsd;
```

**Comment by domenkozar** [comment]
@GrahamcOfBorg build hashdeep

**Pull request merged**